shell/platform/darwin/macos/framework/Source/FlutterView.mm

```diff
@@ -20,14 +33,73 @@ - (instancetype)initWithFrame:(NSRect)frame
   if (self) {
     self.openGLContext = [[NSOpenGLContext alloc] initWithFormat:shareContext.pixelFormat
                                                     shareContext:shareContext];
+
+    [self setWantsLayer:YES];
+
+    // Layer for content (which will be set by surfaceManager). This is separate from
+    // self.layer because it needs to be flipped vertically (using layer.sublayerTransform)
+    contentLayer = [[CALayer alloc] init];
+    [self.layer addSublayer:contentLayer];
+
+    resizeSynchronizer = [[FlutterResizeSynchronizer alloc] initWithDelegate:self];
+    surfaceManager = [[FlutterSurfaceManager alloc] initWithLayer:contentLayer
+                                                    openGLContext:self.openGLContext];
+
     _reshapeListener = reshapeListener;
-    self.wantsBestResolutionOpenGLSurface = YES;
   }
   return self;
 }
 
+- (void)resizeSynchronizerCommit:(FlutterResizeSynchronizer*)synchronizer {
+  [self.openGLContext makeCurrentContext];
+  glFlush();
+  [NSOpenGLContext clearCurrentContext];
+
+  [CATransaction begin];
+  [CATransaction setDisableActions:YES];
+  self.layer.frame = self.bounds;
+  self.layer.sublayerTransform = CATransform3DTranslate(CATransform3DMakeScale(1, -1, 1), 0,
```

Comment:
Maybe add a comment explaining why this transform is needed?

Reply:
I thought about this, and the transform is basically needed because we're rendering to an OpenGL texture, which are flipped. The fact, that we're rendering to an OpenGL texture / IOSurface is an implementation detail of FlutterSurfaceManager, and thus the surface manager should take care of flipping the layer, just like it takes care of settings layer contents. I updated the behavior [here](https://github.com/flutter/engine/pull/21525/commits/19dfde4f63d8aad3a38317054ecf58dafe493dce).